### ISSUE TITLE
Milestone 3 feedback

### ISSUE THREAD

**Issue by Iadel95** [ubco-W2022T1-cosc301/project-group23#1]
# Group23 Feedback

Update on the process: we decided to give everybody feedback on their Milestone 3, and then give you all an opportunity to re-submit after applying the feedback. 

There are some items here that will be noted as "for Milestone 5" - for those items, we don't expect you to complete that until Milestone 5.
For **all** other feedback items, you should apply the feedback and resubmit on Gradescope (we will reopen the submission page later this week).

You should take about one week from when you receive your feedback to resubmit your milestone, This is a hard deadline.

## Group feedback
-Make sure you run a spell check to catch spelling errors, for e.g, this one: https://github.com/jupyterlab-contrib/spellchecker 

## Analysis 1 (Mel)

- Good Job

### Major Feedback 

- (for Milestone 5)  

- your method chaining is not done properly:
    - make it a .py not a .ipynb file
    - check the professor's notes on how to properly implement method chaining.

### Minor Feedback

- Clear your warnings.

- the Pair plot can probably be adjusted to show less data (fewer columns), not everything needs ot be shown here, and there are some duplicates. Redundancies in the pair plot, suggest only showing one half of the pair plot. Try it by setting `corner=True`: https://stackoverflow.com/questions/34087126/plot-lower-triangle-in-a-seaborn-pairgrid

### Points of Pride

- Clean and organized notebook
- good use of markdown and comments to explain your work.

## Analysis 2 (Ella)

- Good job.

### Major Feedback 

- (for Milestone 5)  

- No Major feedback.

### Minor Feedback

- the Pair plot can probably be adjusted to show less data (fewer columns), not everything needs ot be shown here, and there are some duplicates. Redundancies in the pair plot, suggest only showing one half of the pair plot. Try it by setting `corner=True`: https://stackoverflow.com/questions/34087126/plot-lower-triangle-in-a-seaborn-pairgrid

- I see some visualizations present, but there is no analysis done and no explanation for what your graphs **mean**.
You must analyze the visualizations with written statements. Answering the research question should be the primary goal of the project.

### Points of Pride

- Clean and organized notebook
- good use of markdown and comments to explain your work.

## Analysis 3 (Minju)

- Good job.

### Major Feedback 

- (for Milestone 5)  

- No major feedback.

### Minor Feedback

- your histogram plot is not scaled properly. please fix your plots.

- Graphs missing titles and/or axis labels. This makes it hard to know what the graph is trying to represent.

- I see some visualizations present, but there is no analysis done and no explanation for what your graphs **mean**.
You must analyze the visualizations with written statements. Answering the research question should be the primary goal of the project.

### Points of Pride

- Clean and organized notebook
- good use of markdown and comments to explain your work.

**Comment by ESerber** [comment]
@Iadel95 I thought I included brief explanations of my plots in the EDA section using comments, should I move the comments to markdown cells? Is it okay that my explanations are brief for the EDA plots since they are not all used in the actual final analysis? Thank you.

**Comment by Iadel95** [comment]
@ESerber Yes, it is a best practice to use markdowns in python notebooks to explain your work. Think of it as you are writing a report at the same time as implementing your code. Even if your plots are not used in the final analysis, you have plotted them to show you some insights and information. So, you are expected to communicate your insights and thought process. otherwise, the plots would be useless and they shouldn't be there.

**Comment by ESerber** [comment]
@Iadel95 Okay, thank you. I will remove some of the unnecessary plots and move my analysis comments to markdown cells.

**Comment by Iadel95** [comment]
@ESerber I wont recommend removing any plots that helped you to adjust your way of thinking. these are helpful for your thought process and shows the areas that you have explored. What I recommend to do is to write a couple of sentences, explaining what insights did you get from that graph.

**Comment by ESerber** [comment]
@Iadel95 Is it okay if the explanation is briefer than the ones in the final analysis? For example, if I have the same plots in EDA and task 4, should I include the long descriptions in EDA as well or just a brief explanation of what the plot shows and how it may or may not be useful for answering the RQ? Thank you.

**Comment by Iadel95** [comment]
Yes, that would be fine.

**Comment by ESerber** [comment]
Hi @Iadel95, I have done the revisions but the Gradescope assignment does not seem to be open to resubmit. Do you know if it is open (and I'm just not seeing it) or if it hasn't been opened yet? 

I am also having difficulty figuring out the spellchecker and was not aware we were supposed to use something like that (to my knowledge, it had never been mentioned in class or the instructions). I have tried following the link you attached and doing my own searching but I can't seem to figure it out.
